### PR TITLE
Fix #259: Implement limit for max number of errors in process

### LINKED ISSUE #259: Implement limit for max number of errors in process
The following errors are counted:
- failed activation OTP verification (1)
- failed document verification (1)
- user submitted document rejected (2)
- failed selfie verification / OTP (2)
- restart of identity verification (3)

When the max. number of errors is exceeded (default 15), the onboarding process should fail.

Use different weights for different items, so that more important errors stop the process earlier.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationResetService.java [hunk 1]
```diff
@@ -42,21 +50,24 @@ public class IdentityVerificationResetService {
...
     /**
      * Service constructor.
      * @param activationFlagService Activation flag service.
      * @param identityVerificationLimitService Identity verification limit service.
+     * @param processLimitService Onboarding process limit service.
+     * @param processService Common onboarding process service.
      */
     @Autowired
-    public IdentityVerificationResetService(ActivationFlagService activationFlagService, IdentityVerificationLimitService identityVerificationLimitService) {
+    public IdentityVerificationResetService(ActivationFlagService activationFlagService, IdentityVerificationLimitService identityVerificationLimitService, CommonProcessLimitService processLimitService, OnboardingProcessRepository onboardingProcessRepository, CommonOnboardingService processService) {
```
Comment (petrdvorak) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationResetService.java:66]: `OnboardingProcessRepository onboardingProcessRepository` seems unused.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationResetService.java [hunk 2]
```diff
@@ -92,4 +105,37 @@ public void resetIdentityVerification(OwnerId ownerId) throws RemoteCommunicatio
         }
     }
 
+    /**
+     * Update activation flags for reset of identity verification.
+     * @param ownerId Owner identification.
+     * @throws PowerAuthClientException Thrown when communication with PowerAuth server fails.
+     */
+    private void updateActivationFlagsForReset(OwnerId ownerId) throws PowerAuthClientException {
+        List<String> activationFlags = activationFlagService.listActivationFlags(ownerId);
```
Comment (petrdvorak) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationResetService.java:114]: `final`

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationResetService.java [hunk 3]
```diff
@@ -92,4 +105,37 @@ public void resetIdentityVerification(OwnerId ownerId) throws RemoteCommunicatio
...
+
+    /**
+     * Handle a process which just failed due to error score limit.
+     * @param ownerId Owner identification.
+     * @throws OnboardingProcessLimitException Exception thrown due to failed process.
+     * @throws PowerAuthClientException Thrown when communication with PowerAuth server fails.
+     */
+    private void handleFailedProcess(OwnerId ownerId) throws OnboardingProcessLimitException, PowerAuthClientException {
+        // Remove flag VERIFICATION_IN_PROGRESS
+        List<String> activationFlagsToRemove = Collections.singletonList(ACTIVATION_FLAG_VERIFICATION_IN_PROGRESS);
```
Comment (petrdvorak) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationResetService.java:135]: `final`

**Review by petrdvorak — APPROVED** [review]
Looks OK, I see one probably unused variable and two nitpicks...

**Review by saalistaja — APPROVED** [review]
Looks perfect.

**Pull request merged**